src/attribute_type.rs

```diff
@@ -652,12 +652,12 @@ impl fmt::Display for SdpAttributeFmtpParameters {
                 maybe_print_bool_param("usedtx", self.usedtx, false),
                 maybe_print_bool_param("stereo", self.stereo, false),
                 maybe_print_bool_param("useinbandfec", self.useinbandfec, false),
-                maybe_print_bool_param("cbr", self.cbr, false)
+                maybe_print_bool_param("cbr", self.cbr, false),
+                maybe_vector_to_string!("{}", self.encodings, "/"),
+                maybe_print_param("", self.dtmf_tones.clone(), "".to_string()),
+                maybe_vector_to_string!("{}", self.unknown_tokens, ",")
             ]
-            .join(";"),
-            red = maybe_vector_to_string!("{}", self.encodings, "/"),
-            dtmf = maybe_print_param("", self.dtmf_tones.clone(), "".to_string()),
-            unknown = maybe_vector_to_string!("{}", self.unknown_tokens, ",")
```

Comment:
I noticed this too. But I was not sure if there are any spec reasons for keeping these separate. Did you have a chance to check RFCs that this doesn't break something?
It looks like red and dtmf are special in that they don't follow the usual name=value convention.

Reply:
We should split this struct into an enum, which would require a version bump. If it is alright with you, I think we should just fix the missing semi-colon in this patch.

Reply:
After you found out that dmtf and red can't be combined with other params, does still make sense to move them into the comma separated list of params?
I would have thought the most straight forward patch would be to only put the unknown_tokens into the semicolon list, but leave dtmf and red outside. And take care of them when we break them out into an enum like you suggested.